OKR/PACKAGE-MANAGERS.md

```diff
@@ -2,6 +2,224 @@
+
+#### Alternatives
+
+These are some alternative initiatives that _could_ have addressed the KRs.
+
+* Add/Update: A stand-alone IPFS "dropbox" tool. We could write an entirely new tool that simply makes a directory available to the IPFS network using the filestore. Files would be automatically added/removed from IPFS as they're added/removed from the directory.
+  * Pro: Many users would love this kind of tool.
+  * Con: Could be a _lot_ of work.
+  * Con: We want mount support and metadata _anyways_.
+* Update: Out-of-band metadata: Instead of storing timestamps in files to enable rsync, we could store them out-of-band.
+  * Pro: Better deduplication.
+  * Con: Out-of-band data can confuse users (it won't be replicated along with the files).
+  * Con: We need to store some metadata (modes, etc.) anyways.
+  * Con: Some package managers expect files to have accurate timestamps (TODO: check this).
+  * Meh: We're planning on storing metadata in directories, not files. Directories are likely to be unique.
+* Update: A special-purpose rsync tool: We could add a special-purpose `ipfs rsync` command to sync a directory in IPFS with a directory outside of IPFS.
+  * Pro: Pretty simple to implement.
+  * Con: Unlike mount, the files would need to live in IPFS and outside IPFS.
+  * Con: To make this work without timestamps inside IPFS, we'd need the user to pass some kind of "add if modified since" date. That's pretty poor UX.
+
+### Objective: Engage with the package manager community to demonstrate how IPFS meets their needs.
+
+While simply working towards our goal to integrate IPFS and package managers will help drive improvements to the protocol and core implementations, the end goal is to actually _integrate_ IPFS and package managers. To that end, we can't stop at improving the core implementations; we need to demonstrate those improvements to the community.
+
+#### Key Results
+
+The key results for this objective mirror the top three key results in the previous objective.
+
+##### P1: Update Performance
+
+Communicate update performance improvements to the community.
+
+##### P1: Communicate fetch performance improvements
+
+Communicate fetch performance improvements to the community.
+
+##### P3: Communicate add performance improvements
+
+Communicate add performance improvements to the community.
+
+#### Initiatives
+
+In each of these initiatives, we plan to pair with a user/community member.
+
+##### Demo: OpenSuse + Mount
+
+**Owner:** @djdv
+
+Ship a demo of managing an RPM (YaST) repository directly via a mounted IPFS filesystem.
+
+##### Blog: Docker over IPFS
+
+**Owner:** @dirkmc (and friends)
+
+Ship a blog post about speeding up CI by using go-ipfs to distribute docker images.
+
+(docker _is_ a package manager, or close enough)
+
+##### Demo: Demonstrate performance improvements adding NPM to IPFS
+
+**Owner** @aschmahmann/@achingbrain
+
+Ship a demo of adding NPM to IPFS.
```

Comment:
other ideas: time to mirror ipfs-wikipedia (a data package made up of many small files), apt-on-ipfs and clojars-on-ipfs will really focus the work here

Reply:
I'm fine with whatever demo we decide is most effective. As I mentioned in my IPNS comment above we should figure out what the main "this is cool" factors are aside from it just working on IPFS.

As some of our research docs https://github.com/ipfs/package-managers/tree/master/docs (including https://app.mural.co/t/protocollabs6957/m/protocollabs6957/1557168696127/577c9453a3c51199c8163cf0fe5701294e55f99b) indicate there are a number of things we could focus on. They might be things like faster downloads of packages, or ensuring that specific package versions cannot be deleted. Adding in offline operation might be an interesting thing to demonstrate as well.

Reply:
The goal of this initiative is to demo the performance improvements to `ipfs add`.